### ISSUE TITLE
get client's certificate from server's TlsStream?

### ISSUE THREAD

**Issue by udpip7** [sfackler/rust-native-tls#48]
Is there a way to get the client's certificate from a `native_tls::TlsStream` on the server's side?  I couldn't find a method to do so.  It seems like there should be a 'peer_certificate' accessor or something like that as there is on `openssl::SslRef` (available via `openssl::SslStream::ssl`).

**Comment by sfackler** [comment]
You're right that this isn't exposed right now but it should be relatively straightforward to do so.

**Comment by mmstick** [comment]
And this would also be useful to [Concurr](https://github.com/mmstick/concurr) as an authentication method.

**Comment by mikedilger** [comment]
This would also help me with debug conditions around #58

**Comment by chocol4te** [comment]
Is this working? I am currently getting `None` returned from `peer_certificate()` when called on the server side, even with both the client and server identities set, and a root certificate on the client.

**Comment by sfackler** [comment]
Which OS are you on?

**Comment by chocol4te** [comment]
MacOS 10.13.6, I'm trying on a Ubuntu docker image now

**Comment by chocol4te** [comment]
No dice, same result

**Comment by sfackler** [comment]
I'd look into how that test case differs from your own code.

**Comment by chocol4te** [comment]
The test case successfully calls `peer_certificate()` on a client stream, and asserts that it is `None` when called on the server stream, but I am trying to get `Some(client_cert_here)`. Is this not possible?

**Comment by sfackler** [comment]
Ah - the server isn't configured to request the client's certificate, so it isn't going to send it.

**Comment by chocol4te** [comment]
Oh okay, is there any way to change the configuration? Thanks for you help :)

**Comment by sfackler** [comment]
I don't believe that's currently exposed but it's a reasonable thing to have!

**Comment by chocol4te** [comment]
I'll start work on a PR :)
